### PR TITLE
Throttle "Feed parsing" requests

### PR DESCRIPTION
**Pull request by sangwoo108**
When a site has a lot of rss feed urls and the workload is quite heavy, browser process starves and CPU spikes. In order to avoid that, throttle parsing requests with task queue.

Resolves https://github.com/brave/brave-browser/issues/29438

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by sangwoo108** [comment]
@fallaciousreasoning , this fixes the problem on the site, but tests fail 😥

File: components/brave_news/browser/direct_feed_controller.h [hunk 1]
```diff
@@ -97,9 +112,24 @@ class DirectFeedController {
...
+      std::vector<mojom::FeedSearchResultItemPtr> results);
+
   raw_ptr<PrefService> prefs_;
   SimpleURLLoaderList url_loaders_;
+
+  // TODO(sko) We should have a way to cancel requests.
+  // e.g. Navigate to different sites, quit app.
+  // Witthout that, some heavy RSS feed parsing work will prevent new feeds from
+  // detection and app from shutdown.
+  std::queue<FindFeedRequest> pending_requests_;
```
Comment (fallaciousreasoning) [inline components/brave_news/browser/direct_feed_controller.h:127]: This seems like it's probably going to be quite important :smile:

**Review by fallaciousreasoning — APPROVED** [review]
LGTM

File: components/brave_news/browser/direct_feed_controller.h (hunk 1 above)
Comment (sangwoo108) [reply components/brave_news/browser/direct_feed_controller.h:127]: Yeah, It'd be good to find out how to kill worker thread too.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.52.15 Chromium: 112.0.5615.49 (Official Build) nightly (64-bit)
-- | --
Revision | bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}
OS | Windows 11 Version 22H2 (Build 22621.1413)
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/29438#issue-1648990499, reproduced the issue using `1.52.9 Chromium: 112.0.5615.49` as per the following:

**`Example`** | **`Example`**
---------------|-----------------
RegCPUusage | RegperformanceTask

Using `1.52.15 Chromium: 112.0.5615.49`, loaded the same webpage mentioned via https://github.com/brave/brave-browser/issues/29438#issue-1648990499 and ensured that the performance improved as per the following:

**`Example`** | **`Example`**
---------------|-----------------
FixedCPUusage | FixedperformanceTask

**Pull request merged**